nupic/research/frameworks/pytorch/cl_model_utils.py

```diff
@@ -0,0 +1,205 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2020, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+"""
+This module is similar to model_utils.py, except specific for training and evaluating
+in continual learning scenarios.
+"""
+
+import time
+
+import torch
+import torch.nn.functional as F
+
+
+def train_cl_model(
```

Comment:
Do we need these extra functions? If it is just a one-line change to switch the heads, can't this be a callback, or done prior to calling train model?

Reply:
You're right, we can add this one line to `train_model` in model_utils.py, but just beware that function is slowly becoming hard to read/understand since there are so many options ([see here](https://github.com/numenta/nupic.research/blob/master/nupic/research/frameworks/pytorch/model_utils.py#L37-L50)).